### PR TITLE
Tb esp8266

### PR DESCRIPTION
**Pull request by quevedo**
* Minor fixes to the base ndn-cpp library
* Add TCP transport support for Esp8266

### CODE REVIEW

File: include/ndn-cpp/lite/transport/esp8266-tcp-transport-lite.hpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+/**
+ * Copyright (C) 2015-2017 Regents of the University of California.
+ * @author: Jeff Thompson <jefft0@remap.ucla.edu>
```
Comment (jefft0) [inline include/ndn-cpp/lite/transport/esp8266-tcp-transport-lite.hpp]: Can you add yourself as an @author?

File: src/lite/transport/esp8266-tcp-transport-lite.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+/**
+ * Copyright (C) 2015-2017 Regents of the University of California.
+ * @author: Jeff Thompson <jefft0@remap.ucla.edu>
```
Comment (jefft0) [inline src/lite/transport/esp8266-tcp-transport-lite.cpp]: Can you add yourself as an @author?

File: src/c/util/crypto.c [hunk 3]
```diff
@@ -22,13 +22,16 @@
...
+ndn_Error
+ndn_generateRandomBytes(uint8_t *buffer, size_t bufferLength)
 {
   // Assume the application has already initialized it, e.g.:
   // randomSeed(analogRead(0));
   size_t i;
   for (i = 0; i < bufferLength; ++i)
     buffer[i] = random(0, 256);
+
+  return NDN_ERROR_success;
```
Comment (jefft0) [inline src/c/util/crypto.c]: Thanks for catching this!

File: include/ndn-cpp/lite/transport/esp8266-tcp-transport-lite.hpp (hunk 1 above)
Comment (quevedo) [inline include/ndn-cpp/lite/transport/esp8266-tcp-transport-lite.hpp]: I have added myself as an @author

File: src/lite/transport/esp8266-tcp-transport-lite.cpp (hunk 2 above)
Comment (quevedo) [inline src/lite/transport/esp8266-tcp-transport-lite.cpp]: I have added myself as an @author

**Pull request merged**